### PR TITLE
Spring cleaning

### PR DESCRIPTION
**Pull request by ConnorNeed**
The clang tidy isn't working how I like yet but I will work on it.

The first time install script works (Because I got to test it with the jetson)

The make.sh script works

the clang format script is my personal one I have developed over the years. It's pretty close to google standard with some extra alignment features. I am open to suggestions on it.

Cleaned up some unneeded files.

Updated the readme

Fixed some rosdep dependancies

Unf*cked the jetson in the progress 

I forgot to push the format changes so that will have to wait till I get back to the bay or just run the make script then push all the changes to have all the format changes pushed.

Just using black's default formatting. If we don't like it then we can make a config file (I might be a snob when it comes to c++ formating but I don't have as strong of feelings when it comes to python) C++ rules. I hate snakes

(I need a life, this is a cry for help)

### CODE REVIEW

File: firstTimeInstall.sh
Comment (ConnorNeed) [inline firstTimeInstall.sh:1]: I'll make all the apt installs a single command. Will technically run a hair faster and look cleaner

File: launch/current_pose_tester_launch.py
Comment (ConnorNeed) [inline launch/current_pose_tester_launch.py:1]: This was Will's. I talked to him and he said it wasn't useful and could go

**Review by ChrisRusu1 — APPROVED** [review]
Looks good to me, I wont merge it yet since I want to see what Mat thinks about the first time install and the make script. Also about the formatting I think its fine to make this change with the clang and black formatting and then do another commit later running the formatting and cleaning up the code.
I think I also have some local changes on the jetson to add some comments but they arnt fully done yet so we can run the script and then get those in for when people will start doing the controller project

File: firstTimeInstall.sh
Comment (noodlekid) [inline firstTimeInstall.sh]: if members run this more than once this is going to get called and make .bashrc look like shit

Maybe modify bash script to include something like this

`if ! grep -q "source /opt/ros/humble/setup.bash" ~/.bashrc; then
  echo "source /opt/ros/humble/setup.bash" >> ~/.bashrc
  echo "ROS 2 sourced in bashrc"
fi`

File: firstTimeInstall.sh
Comment (noodlekid) [inline firstTimeInstall.sh]: Could use some `-y` here and there to avoid user prompts

File: make.sh
Comment (noodlekid) [inline make.sh:1]: lgtm

File: make.sh
Comment (noodlekid) [inline make.sh:57]: could avoid adding `|| exit 1` with a `set -e` to the head of the script to exit on any command fail.

**Review by noodlekid — CHANGES_REQUESTED** [review]
LGTM consider the comments if you please

**Review by noodlekid — APPROVED** [review]
lgtm 

sent from Mateusz's iPhone

File: firstTimeInstall.sh
Comment (ConnorNeed) [reply firstTimeInstall.sh]: Nice addition

File: firstTimeInstall.sh
Comment (ConnorNeed) [reply firstTimeInstall.sh]: Yeah I was torn on whether to use it or not. It installs a lot of stuff so I had left in the prompt so new members got told how much space it was going to take up. I don't really care either way.

File: make.sh
Comment (ConnorNeed) [reply make.sh:57]: You stumped onto a very controversial one here. Set -e is considered by some to be bad practice. Not every program returns zero even if it acts as expected. I didn't want to handicap what we can use because it has to return zero. 

I think this script currently should be fine to use set -e as is but it might have to be switched back in the future.

This article explains it better than I can.
https://mywiki.wooledge.org/BashFAQ/105

**Pull request merged**
